src/AppBundle/Entity/Tag.php

```diff
@@ -66,4 +73,45 @@ public function __toString()
     {
         return $this->name;
     }
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->posts = new \Doctrine\Common\Collections\ArrayCollection();
+    }
+
+    /**
+     * Add post
+     *
+     * @param \AppBundle\Entity\Post $post
+     *
+     * @return Tag
+     */
+    public function addPost(\AppBundle\Entity\Post $post)
```

Comment:
this adder is worse than anything else you did, because it will **not** add the post (as you don't update the owning side of the relation)